examples/feature_selection/plot_rfe_digits.py

```diff
@@ -3,34 +3,153 @@
+# a baseline accuracy to compare against.
+
+# %%
+from sklearn.metrics import accuracy_score
+from sklearn.svm import SVC
+
 svc = SVC(kernel="linear", C=1)
-rfe = RFE(estimator=svc, n_features_to_select=1, step=1)
-rfe.fit(X, y)
-ranking = rfe.ranking_.reshape(digits.images[0].shape)
+svc.fit(X_train, y_train)
+y_pred = svc.predict(X_test)
+accuracy_all_features = accuracy_score(y_test, y_pred)
+
+print(f"Accuracy using all {X_train.shape[1]} features: {accuracy_all_features:.4f}")
+
+# %%
+# Feature Selection with RFE
+# --------------------------
+#
+# Now, we'll employ :class:`~sklearn.feature_selection.RFE` to select the 10 most
+# discriminative features. The goal is to determine if a reduced set of important
+# features can either maintain or even improve the classifier's performance.
+
+# %%
+from sklearn.feature_selection import RFE
+from sklearn.pipeline import Pipeline
+
+num_features_to_select = 10
+
+# Create a pipeline with feature selection followed by SVM
+pipe = Pipeline(
+    [
+        (
+            "rfe",
+            RFE(
+                estimator=SVC(kernel="linear", C=1),
+                n_features_to_select=num_features_to_select,
+            ),
+        ),
+        ("svc", SVC(kernel="linear", C=1)),
+    ]
+)
+
+# Fit pipeline to the data
+pipe.fit(X_train, y_train)
+
+# Compute predictions on test data
+y_pred_rfe = pipe.predict(X_test)
+
+# Get accuracy of model using selected features
+accuracy_selected_features = accuracy_score(y_test, y_pred_rfe)
+
+print(
+    f"Accuracy using {num_features_to_select} selected features:"
+    f" {accuracy_selected_features:.4f}"
+)
+
+# %%
+# Feature Selection with RFECV
+# ----------------------------
+#
+# To search over all possible numbers of features, we'll employ
+# :class:`~sklearn.feature_selection.RFECV`.
+
+# %%
+from sklearn.feature_selection import RFECV
+
+# Choose estimator
+estimator = SVC(kernel="linear")
+
+min_features_to_select = 1
+
+# Define RFECV object
+rfecv = RFECV(
+    estimator,
+    step=1,
+    min_features_to_select=min_features_to_select,
+    cv=5,
+    scoring="accuracy",
+    n_jobs=-1,
+)
+
+# Fit to the data
+rfecv.fit(X, y)
+
+# Extract the optimal number of features from the best estimator
+optimal_num_features = rfecv.n_features_
+
+print(f"Optimal number of features: {optimal_num_features}")
+
+# %%
+# Visualizing Feature Importance after RFE
+# ----------------------------------------
+#
+# RFECV and RFE provide a ranking of the features based on their importance.
+# We can visualize this ranking to gain insights into which pixels
+# (or features) are deemed most significant by RFECV in the digit
+# classification task.
 
-# Plot pixel ranking
+# %%
+ranking = rfecv.ranking_.reshape(digits.images[0].shape)
 plt.matshow(ranking, cmap=plt.cm.Blues)
 plt.colorbar()
-plt.title("Ranking of pixels with RFE")
+plt.title("Ranking of pixels with RFECV")
 plt.show()
```

Comment:
According to the documentation `ranking_` is an array where the most important features are assigned rank 1 and the higher ranking the less important. Notice that in the current version of the example you have shades of blue going from 1 to 64 (we have 8 x 8 pixels) whereas your code uses a model which already truncated the feature space to keep only the 42 most relevant pixels and degenerating the rest to a value of 1. I am afraid this was not the spirit of the example. You can alternatively show the feature importance of the whole 64 pixels using different seeds of the `train_test_split` to check for stability.